beetmoverscript/src/beetmoverscript/gcloud.py

```diff
@@ -72,7 +76,21 @@ def handle_exception(e):
         handle_exception(e)
         return
     log.info(f"Found GCS bucket {bucket} - proceeding with GCS uploads.")
-    context.gcs_client = client
+    return client
+
+
+def _get_artifact_registry_client(context, product):
+    """Set up a google-cloud-artifact-registry client"""
+    client = artifactregistry_v1.ArtifactRegistryAsyncClient()
+    return client
+
+
+def set_gcp_client(context):
+    product = get_product_name(context.task, context.config)
+    if "repo" in context.resource_type:
+        context.gcp_client = _get_artifact_registry_client(context, product)
+    else:
+        context.gcp_client = _get_gcs_client(context, product)
```

Comment:
`gcp_client` has very different methods when dealing with the different services.  `context` is bad enough as a god object, and we made this mistake with "bucket name" property already.

I suggest we split it into `gcs_client` and `gar_client` (or something of the sorts).

```suggestion
        context.gar_client = _get_artifact_registry_client(context, product)
        ....
        context.gcs_client = _get_gcs_client(context, product)
```